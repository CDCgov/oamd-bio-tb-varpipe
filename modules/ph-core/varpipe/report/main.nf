process VARPIPE_REPORT {

    tag "${meta.id}"
    label 'process_single'

    container 'oamd-bio-biocontainers-python:3.12_80c558c_v3'

    input:
    tuple val(meta), path(statsFile), path(targetRegionCoverage), path(drLociFinalAnnotation)

    output:
    tuple val(meta), path("summary.txt"), emit: summary
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'create_report.py'

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch summary.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
