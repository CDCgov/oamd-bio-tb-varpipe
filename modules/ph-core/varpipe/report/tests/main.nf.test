nextflow_process {

    name "Test Process VARPIPE_REPORT"
    script "../main.nf"
    process "VARPIPE_REPORT"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/report"

    test("varpipe create report [stats] [coverage] [final annotation]") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_stats.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_target_region_coverage.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_DR_loci_Final_annotation.txt')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe create report [stats] [coverage] [final annotation] -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_stats.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_target_region_coverage.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_DR_loci_Final_annotation.txt')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
