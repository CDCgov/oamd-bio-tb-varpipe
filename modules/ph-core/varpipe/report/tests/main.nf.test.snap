{
    "varpipe create report [stats] [coverage] [final annotation] -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,fde7f2733dc38aab60c37aac0f593128"
                ],
                "summary": [
                    [
                        {
                            "id": "test"
                        },
                        "summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fde7f2733dc38aab60c37aac0f593128"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T16:50:39.868126625"
    },
    "varpipe create report [stats] [coverage] [final annotation]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "summary.txt:md5,46d4a09785c12d3a149dacfd6a2481d1"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d257db12c9c7e91617ec6de19e02d8c4"
                ],
                "summary": [
                    [
                        {
                            "id": "test"
                        },
                        "summary.txt:md5,46d4a09785c12d3a149dacfd6a2481d1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d257db12c9c7e91617ec6de19e02d8c4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T16:50:32.078534414"
    }
}
